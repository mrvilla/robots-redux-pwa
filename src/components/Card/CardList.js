import React from 'react';
import Card from './Card';

const CardList = ({ robots }) => {
  // if (true) {
    // throw new Error('Oh Shit! Sorry bruh');
  // }

  return (
    <div className='tc'>
      {robots.map((user, idx) => {
        return <Card
          key={idx}
          id={robots[idx].id}
          name={robots[idx].name}
          email={robots[idx].email}
        />
      })}
    </div>
  );
};

export default CardList;
