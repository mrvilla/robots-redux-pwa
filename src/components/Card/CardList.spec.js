import React from "react";
import renderer from "react-test-renderer";
import CardList from './CardList';

describe("CardList Snapshot testing", () => {
  it("CardList renders correctly", () => {
    const mockRobots = [
      {
        id: 1,
        name: 'John Doe',
        username: 'john_doe',
        email: 'j.doe@mail.com'
      }
    ]
    const domTree = renderer.create(<CardList robots={mockRobots} />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
