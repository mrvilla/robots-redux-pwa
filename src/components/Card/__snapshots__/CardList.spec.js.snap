// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList Snapshot testing CardList renders correctly 1`] = `
<div
  className="tc"
>
  <div
    className="dib br3 pa2 ma2 tc"
  >
    <img
      alt="robot"
      src="https://robohash.org/1?size=220x220"
    />
    <div>
      <h2>
        John Doe
      </h2>
      <p>
        j.doe@mail.com
      </p>
    </div>
  </div>
</div>
`;
